spec/robots/accession/publish_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+require File.expand_path(File.dirname(__FILE__) + '/../../../robots/accession/publish')
+
+describe Robots::DorRepo::Accession::Publish do
+  let(:druid) { 'druid:oo000oo0001' }
+
+  it 'includes behavior from LyberCore::Robot' do
+    robot = Robots::DorRepo::Accession::Shelve.new
+    expect(robot.methods).to include(:work)
+  end
+
+  it 'calls .publish_metadata if that method is defined (e.g. item)' do
+    object = double(:publish_metadata => true)
+    expect(object.respond_to?(:publish_metadata)).to be true
+    expect(Dor).to receive(:find).with(druid).and_return(object)
+    robot = Robots::DorRepo::Accession::Publish.new
+    expect(object).to receive(:publish_metadata)
+    robot.perform(druid)
+  end
+
+  it 'does not call .publish_metadata if that method is not defined (e.g. apo)' do
+    object = Object.new
+    expect(object.respond_to?(:publish_metadata)).to be false
+    expect(Dor).to receive(:find).with(druid).and_return(object)
+    robot = Robots::DorRepo::Accession::Publish.new
```

Comment:
Joe and I realized this won't work, since expect apparently defines the mock method you are testing (:publish_metadata) to confirm it won't get called...but then our logic thinks it exists, and calls it, causing the test to fail... it took us a while to sort this out, which means you can't test the expectation like that and instead needs to test via a non-exception